aiida/manage/configuration/__init__.py

```diff
@@ -313,46 +311,22 @@ def load_documentation_profile():
         transports,
     )
     from aiida.cmdline.params import arguments, options  # pylint: disable=unused-import
-    from aiida.storage.psql_dos.models.base import get_orm_metadata
-
-    from .config import Config
-
-    global CONFIG
-
-    with tempfile.NamedTemporaryFile() as handle:
-        profile_name = 'readthedocs'
-        profile_config = {
-            'storage': {
-                'backend': 'core.psql_dos',
-                'config': {
-                    'database_engine': 'postgresql_psycopg2',
-                    'database_port': 5432,
-                    'database_hostname': 'localhost',
-                    'database_name': 'aiidadb',
-                    'database_password': 'aiidadb',
-                    'database_username': 'aiida',
-                    'repository_uri': 'file:///dev/null',
-                }
+    from aiida.storage.sqlite_temp import SqliteTempBackend
+    
+    profile = load_profile(
+        SqliteTempBackend.create_profile(
+            'readthedocs',
+            options={
+                'warnings.development_version': False,
             },
-            'process_control': {
-                'backend': 'rabbitmq',
-                'config': {
-                    'broker_protocol': 'amqp',
-                    'broker_username': 'guest',
-                    'broker_password': 'guest',
-                    'broker_host': 'localhost',
-                    'broker_port': 5672,
-                    'broker_virtual_host': '',
-                }
-            },
-            'options': {
-                'warnings.showdeprecations': False
-            }
-        }
-        config = {'default_profile': profile_name, 'profiles': {profile_name: profile_config}}
-        CONFIG = Config(handle.name, config)
-        load_profile(profile_name)
-
-        # we call this to make sure the ORM metadata is fully populated,
-        # so that ORM models can be properly documented
-        get_orm_metadata()
+            debug=False,
+        ),
+        allow_switch=True,
+    )
+    config = get_config()
+    config.add_profile(profile)
+    config.set_default_profile(profile.name)
+    
+    #    # we call this to make sure the ORM metadata is fully populated,
+    #    # so that ORM models can be properly documented
+    #    get_orm_metadata()
```

Comment:
I am not sure how this is used, I found there is a similar model implementation for `sqlite_temp` storage, how to get orm metadata from it?

Reply:
This would be done implicitly by loading the storage backend, i.e., `get_manager().get_profile_storage()`. But apparently the docs build is passing without it, so it is no longer necessary.

Reply:
I wonder if even need this `load_documentation_profile` at all anymore. This may be a relic from before v2.0 when we still had a Django backend as well which required the database to be loaded on import. This is no longer the case, so I think we can just import without having access to a profile storage. Could you try removing the `load_documentation_profile` call from the `docs/source/conf.py` file and see if the docs still build? If the case, we can just get rid of this function entirely.

Reply:
It is also called by `pre-commit` through `python ./utils/validate_consistency.py verdi-autodocs`, and that is where the verdi doc generated I assume.

Reply:
Yes, but I think you can also remove it there, it doesn't need it anymore I think

Reply:
Sorry, you mean remove `verdi-autodocs` pre-commit hook? My point was the `load_documentation_profile` is appear in both `validate_consistency.py` which called by `verdi-autodocs` pre-commit and in the `conf.py` by sphinx, thus make sense to put it as a function.

Reply:
No I mean remove the `load_documentation_profile` call from `validate_consistency.py` and also from `conf.py`. I don't it is needed anymore in either of these places

Reply:
Seems you still need this part in the `conf.py`:
```python

    # imports required for docs/source/reference/api/public.rst
    from aiida import (  # pylint: disable=unused-import
        cmdline,
        common,
        engine,
        manage,
        orm,
        parsers,
        plugins,
        schedulers,
        tools,
        transports,
    )
    from aiida.cmdline.params import arguments, options  # pylint: disable=unused-import
```
Apparently still need explicit import to have the autoapi work properly.

Reply:
Seems it should be there, otherwise, the doc build will failed with the module not being found.

Reply:
Seems the profile is needed to build the apidoc.

Reply:
I am really confused though, because it is speaking of `process.aiida` which is the temp profile that was created for running the `howto/archive_profile.md` notebook. It is a bit of a coincidence that we just fixed things there. I just merged that PR, so maybe you can rebase this PR and see if it still fails.

Reply:
You are all correct about it. :+1: